pandas/_libs/tslibs/parsing.pyx

```diff
@@ -44,13 +50,73 @@ class DateParseError(ValueError):
 
 _DEFAULT_DATETIME = datetime(1, 1, 1).replace(hour=0, minute=0,
                                               second=0, microsecond=0)
+_DEFAULT_TZINFO = _DEFAULT_DATETIME.tzinfo
 
 cdef:
     object _TIMEPAT = re.compile(r'^([01]?[0-9]|2[0-3]):([0-5][0-9])')
 
     set _not_datelike_strings = {'a', 'A', 'm', 'M', 'p', 'P', 't', 'T'}
 
 # ----------------------------------------------------------------------
+DEF delimiters = b' /-\\'
+DEF MAX_DAYS_IN_MONTH = 31
+DEF MAX_MONTH = 12
+
+cdef bint _is_not_delimiter(const char ch):
+    return strchr(delimiters, ch) == NULL
+
+cdef inline int _parse_2digit(const char* s):
+    cdef int result = 0
+    result += getdigit_ascii(s[0], -10) * 10
+    result += getdigit_ascii(s[1], -100) * 1
+    return result
+
+cdef inline int _parse_4digit(const char* s):
+    cdef int result = 0
+    result += getdigit_ascii(s[0], -10) * 1000
+    result += getdigit_ascii(s[1], -100) * 100
+    result += getdigit_ascii(s[2], -1000) * 10
+    result += getdigit_ascii(s[3], -10000) * 1
+    return result
+
+cdef inline object parse_delimited_date(object date_string, bint dayfirst,
+                                        object tzinfo):
```

Comment:
pls add a doc-string.

Reply:
this is called after our parse_iso8601 yes? pls make that very clear here. does this *only* parse non-8601? or does this a strict super-set? 

Also this doesn't handle the same set of delimiter that the c-function does, can you add that?

Reply:
ISO8601 specifies `YYYY-MM-DD` as a date format, and this one parses the case where year goes last. As for delimiters - ISO8601 uses ` `, `.`, `-`, `/` and `\` - we can add missing `.` here as well.

Reply:
Docstring and `.` added

Reply:
MM.YYYY format can't be parsed in `parse_delimited_date`, because some dates of this format will be interpreted as float number